firmware/boron-ins-fsm/src/im21door.cpp

```diff
@@ -153,30 +153,23 @@ void threadBLEScanner(void *param) {
   BLE.setScanTimeout(5);
 
   while(true){
-
-    int count = BLE.scan(scanResults, SCAN_RESULT_MAX);
-
+    //filter has to be remade each time, as globalDoorId (from eeprom) is usually read while this thread is already in while loop
+    BleScanFilter filter; 
+    char address[sizeof("B8:7C:6F:5C:23:49")]; 
+    sprintf(address, "B8:7C:6F:%02X:%02X:%02X", globalDoorID.byte3, globalDoorID.byte2, globalDoorID.byte1); 
+    filter.deviceName("iSensor ").address(address); 
+
+    //scanning for door sensors of correct id
+    Vector<BleScanResult> scanResults = BLE.scanWithFilter(filter); 
```

Comment:
Thanks for looking into it!
Did you ever see any red blinking lights like Sampath mentioned @BraveKerry ?

What happens if you are in the office with and you open and close a whole bunch of door sensors at the same time?

Reply:
Haven't had red blinking lights yet, although that may be because I haven't explicitly tested for it. I'll try that multiple-door-sensor test at the office.

Reply:
Now that I think about it, multiple door sensors do not work as a test, as the filter also filters by door sensor address id. However, I did test with quickly opening and closing one door sensor, and there were no red blinking lights.

Reply:
Oh! So it really does scan for just a specific, single door sensor. That makes me feel much better about it. Thanks for looking into it!